### PR TITLE
[RW-966][risk=no] Load entire Visits tree

### PR DESCRIPTION
**Pull request by dolbeew**
Loads entire tree initially instead of loading children when each parent node is expanded

### CODE REVIEW

File: ui/src/app/cohort-search/node/node.component.ts [hunk 1]
```diff
@@ -43,28 +51,67 @@ export class NodeComponent implements OnInit, OnDestroy {
...
   }
 
   ngOnDestroy() {
-    this.subscription.unsubscribe();
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  findParent(child, itemList) {
```
[the remark this answers is not in the document]
Comment (dolbeew) [reply ui/src/app/cohort-search/node/node.component.ts]: Good idea. Done!

File: ui/src/app/cohort-search/node/node.component.ts [hunk 2]
```diff
@@ -28,13 +31,18 @@ export class NodeComponent implements OnInit, OnDestroy {
    * `error`: was there an error loading the children of this criterion?
    * `children`: when loaded, this node's children are stored here.
    *
-   * The initial load of the children is deferred until the subtree is first
-   * expanded.
+   * If the root criterion has 'fullTree' set to true, the entire tree is
+   * loaded. Otherwise, the initial load of the children is deferred until
+   * the subtree is first expanded.
+   *
+   * In the future, we may want put full trees in a separate component to
```
[the remark this answers is not in the document]
Comment (dolbeew) [reply ui/src/app/cohort-search/node/node.component.ts]: Thanks!

File: ui/src/app/cohort-search/search-group/search-group.component.ts [hunk 3]
```diff
@@ -39,10 +39,12 @@ export class SearchGroupComponent {
     this.actions.removeGroup(this.role, this.groupId);
   }
 
-  launchWizard(criteriaType: string) {
+  launchWizard(criteria: any) {
```
Comment (freemabd1) [inline ui/src/app/cohort-search/search-group/search-group.component.ts]: Would it make sense to use a type here and make constants.ts use an object with constructor?

Comment (dolbeew) [reply ui/src/app/cohort-search/search-group/search-group.component.ts]: I agree, we should do this in the future. I'll create a ticket for it.

**Pull request merged**